Source/Engine/Utilities/Extensions.cs

```diff
@@ -256,7 +256,7 @@ public static void Shuffle<T>(this Random random, IList<T> collection)
         /// <returns>A random <see cref="byte"/> between min and max.</returns>
         public static byte NextByte(this Random random, byte min = 0, byte max = byte.MaxValue)
```

Comment:
This is tricky, since a byte cannot be larger than `255`, however, the usual convention is to have an *exclusive* upper bound. I guess this part of why C# doesn't have a random byte method with a range as a parameter? 🤔

Reply:
Could you elaborate, because if you look into the definition of the method you can see that I've adjusted the `max` part by incrementing it by one.

Reply:
Yep, I meant that the method's inputs are off by one. 
Like, `random.NextByte(0, 2)` would return 0, 1 or 2.
Meanwhile, `random.Next(0, 2)` would return 0 or 1.

Reply:
Ah alright, should be it hopefully with this commit 😄